### PR TITLE
[PKG-2938] tiktoken 0.5.1 :snowflake:

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Current package info**

| Channel | Downloads | Version | Platforms | License |
| --- | --- | --- | --- | --- |
| conda-forge | [Conda Downloads](https://anaconda.org/conda-forge/tiktoken) | [Conda Version](https://anaconda.org/conda-forge/tiktoken) | [Conda Platforms](https://anaconda.org/conda-forge/tiktoken) | Conda License |

Changelog: https://github.com/openai/tiktoken/blob/0.5.1/CHANGELOG.md
License: https://github.com/openai/tiktoken/blob/0.5.1/LICENSE
Requirements: 
- https://github.com/openai/tiktoken/blob/0.5.1/pyproject.toml
- https://github.com/openai/tiktoken/blob/0.5.1/setup.py

Actions:
1. Clean up the recipe
2. Add `--no-deps --no-build-isolation` to `script`
3. Add `libgcc-ng` on `linux`
4. Add a test suite
5. Add dev & doc urls
6. Add `description` 
7. Add `license_family`

### CODE REVIEW

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 1]
```diff
@@ -6,46 +6,57 @@ package:
...
+    - {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+  missing_dso_whitelist:  # [s390x]
+    - "$RPATH/ld64.so.1"  # [s390x]
 
 requirements:
   build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-    - {{ compiler('cxx') }}
+    - {{ compiler('cxx') }}  # [win]
```
Comment (akabanovs) [inline recipe/meta.yaml]: Isn't it only python and rust, why do we need this?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -6,46 +6,56 @@ package:
...
-    - setuptools >=62.4
-    - wheel
-    - setuptools-rust >=1.5.2
     - pip
+    - setuptools
+    - setuptools-rust
+    - wheel
   run:
     - python
+    - libgcc-ng  # [linux]
```
Comment (akabanovs) [inline recipe/meta.yaml:32]: And do we need this, actually?

Comment (skupr-anaconda) [reply recipe/meta.yaml:32]: Yes, it fails without it.

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Fixed

File: recipe/meta.yaml (hunk 2 above)
Comment (akabanovs) [reply recipe/meta.yaml:32]: Does it mean some c++ run-time dependency in the chain actually requires it?

**Review by akabanovs — APPROVED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:32]: I removed it and added again `{{ compiler('cxx') }}`  but with `# [linux]`

**Comment by akabanovs** [comment]
b3fe6857d410224180cb1d699436322744d13bb6 worked fine. Why adding the last commit?

**Comment by skupr-anaconda** [comment]
> [b3fe685](https://github.com/AnacondaRecipes/tiktoken-feedstock/commit/b3fe6857d410224180cb1d699436322744d13bb6) worked fine. Why adding the last commit?

Sorry, you are right. I've incorrectly interpreted your words. I'll revert my changes

**Pull request merged**